third_party/move/move-vm/runtime/src/storage/loader/eager.rs

```diff
@@ -44,18 +43,23 @@ impl<'a, T> StructDefinitionLoader for EagerLoader<'a, T>
 where
     T: ModuleStorage,
 {
+    fn is_lazy_loading_enabled(&self) -> bool {
+        debug_assert!(!self.runtime_environment().vm_config().enable_lazy_loading);
+        false
+    }
+
     fn load_struct_definition(
         &self,
         _gas_meter: &mut impl GasMeter,
-        _traversal_context: &mut TraversalContext,
+        _traversal_context: &mut dyn ModuleTraversalContext,
         idx: &StructNameIndex,
     ) -> PartialVMResult<Arc<StructType>> {
         let struct_name = self
             .runtime_environment()
             .struct_name_index_map()
             .idx_to_struct_name_ref(*idx)?;
 
```

Comment:
Does it still make sense to pass checking traversal context to eager, since everything should always be loaded? (just for extra invariant checking.. if it can add any value and not degrade perf)

Reply:
Good point, so here I think that it would be useful, but there are places where eager was no covering everything (e.g., metadata of a module) so I ad to make sure invariant is checked for lazy for compatibility. Here we should probably be ok, but I am not sure - so thought that making eager compatible, and lazy safer with invariant checks is the easiest way.

Because if something is missed we need feature gating even for eager loader?

Reply:
Ah, that's a good consideration. But maybe the argument here could be if there is more things missed and how useful that would be to know (now or future) vs the annoyance.